### DISCUSSION TITLE
Allow try without braces

### DISCUSSION THREAD

**Discussion by tonyfaull** [dotnet/csharplang#6192] [Language Ideas] [answered]
I propose an alternative syntax for `try ... catch` blocks which reduces a level of nesting. It takes inspiration from _using declarations_. You would omit the braces from the `try;` block, while retaining them in the `catch`.

For example,
```
public int DoubleSum(int a, int b)
{
    try;

    var n = a + b;
    n *= 2;
    return n;

    catch (Exception ex)
    {
        Console.WriteLine(ex.Message);
    }
}
```

**Comment by 333fred** [comment, marked as answer]
Basically a duplicate of https://github.com/dotnet/csharplang/discussions/908, with only a minor syntax change.

**Reply by tonyfaull** [reply]
Thanks for the feedback. I think the syntax difference is quite significant, so it's not quite a duplicate. This is an improved version. The initial `try;` statement is essential as it emphasizes that this is a form of the `try..catch` statement.

**Comment by HaloFour** [comment]
How does this resolve the ambiguities inherent in attempting to nest `try` statements without the blocks?

**Reply by tonyfaull** [reply]
I agree we would need to restrict usage to avoid ambiguity. For instance we could limit it to one `try;` statement per method. We could further require the `try;` to be at the beginning of the scope and the `catch` block at the end if that makes sense.

**Reply by HaloFour** [reply]
Such limitations make me think that the applicability of such a language feature would be too uncommon to be worthwhile.

**Reply by tonyfaull** [reply]
Perhaps those restrictions are too strict. We could allow multiple `try;` statements in series but not nested. 
e.g.
```
bool Copy(string source, string target)
{
    string fileText;
    try;

    fileText = ReadFile(source);

    catch (FileNotFoundException)
    {
        fileText = "";
    }

    try;

    WriteFile(target, fileText);
    return true;

    catch (ReadOnlyFileException)
    {
        return false;
    }
}
```

**Reply by HaloFour** [reply]
IMO that's much easier to read with the blocks:

```cs
bool Copy(string source, string target)
{
    string fileText;
    try
    {
        fileText = ReadFile(source);
    }
    catch (FileNotFoundException)
    {
        fileText = "";
    }

    try
    {
        WriteFile(target, fileText);
        return true;
    }
    catch (ReadOnlyFileException)
    {
        return false;
    }
}
```

**Reply by tonyfaull** [reply]
I think you're right. The code blocks are more readable. Happy to let this one go.

**Comment by iam3yal** [comment]
The `try` statement with braces is much more readable to me whereas the ` try;` in the OP code looks looks like something that doesn't belong to the language.
